src/components/Table.vue

```diff
@@ -1492,6 +1508,8 @@ export default {
         this.selectAllByPage = selectAllByPage;
       }
 
+      this.selectAllByGroup = Boolean(selectAllByGroup);
```

Comment:
I'm curious why all these `typeof === 'boolean'` are needed? Can't you just cast to Boolean() like I did?